internal/clients/recommendations/client.go

```diff
@@ -34,40 +37,49 @@ func NewClient(conf RecommendationsClientConfig) (*RecommendationsClient, error)
 	if conf.URL == "" {
 		logrus.Warn("Recommendation URL not set, client will fail")
 	}
-	client, err := createClient(conf.URL, conf.CA)
+	client, err := createClient(conf.URL, conf.CA, conf.Proxy)
 	if err != nil {
 		return nil, fmt.Errorf("error creating recommend http client")
 	}
 
 	rc := RecommendationsClient{
 		URL:     conf.URL,
+		Proxy:   conf.Proxy,
 		tokener: conf.Tokener,
 		client:  client,
 	}
 
 	return &rc, nil
 }
 
-func createClient(recommendationsURL string, ca string) (*http.Client, error) {
-	if !strings.HasPrefix(recommendationsURL, "https") || ca == "" {
-		return &http.Client{}, nil
-	}
+func createClient(recommendationsURL string, ca string, proxyURL string) (*http.Client, error) {
+	var transport *http.Transport
 
-	var tlsConfig *tls.Config
-	caCert, err := os.ReadFile(filepath.Clean(ca))
-	if err != nil {
-		return nil, err
-	}
+	if strings.HasPrefix(recommendationsURL, "https") && ca != "" {
+		caCert, err := os.ReadFile(filepath.Clean(ca))
+		if err != nil {
+			return nil, err
+		}
+
+		caCertPool := x509.NewCertPool()
+		caCertPool.AppendCertsFromPEM(caCert)
 
-	caCertPool := x509.NewCertPool()
-	caCertPool.AppendCertsFromPEM(caCert)
+		tlsConfig := &tls.Config{
+			MinVersion: tls.VersionTLS12,
+			RootCAs:    caCertPool,
+		}
 
-	tlsConfig = &tls.Config{
-		MinVersion: tls.VersionTLS12,
-		RootCAs:    caCertPool,
+		transport = &http.Transport{
+			TLSClientConfig: tlsConfig,
+		}
+	} else {
+		transport = &http.Transport{}
 	}
 
-	transport := &http.Transport{TLSClientConfig: tlsConfig}
+	if proxyURL != "" {
+		proxyURLParsed := url.URL{Host: proxyURL, Scheme: "http"}
```

Comment:
Why not use https://pkg.go.dev/net/url#Parse?

Reply:
@croissanne I tried to use proxyURLParsed, err := url.Parse(proxyURL), but it is not working, 
beucase we didnt define the http:// for proxy paramemter at Vault system.

Reply:
ah well I'll update it to include the scheme then.

Reply:
updated!